### PR TITLE
Chat notification

### PR DESCRIPTION
**Pull request by jenessatan**
- add unread field to Messages schema plus associated endpoints
- created notification socket event listeners/emitters
- fix timestamp bug

### CODE REVIEW

File: socket/socket.js [hunk 1]
```diff
@@ -214,6 +239,7 @@ const insertMessages = async (msgData) => {
       fromUserId: msgData.fromUserId,
       toUserId: msgData.toUserId,
       content: msgData.content,
+      date: new Date()
```
Comment (abid-salahi) [inline socket/socket.js]: i think the `isUnread` field is supposed to be added here given it's "required" for a new message in the schema

File: socket/socket.js [hunk 2]
```diff
@@ -94,6 +94,31 @@ const socketEvents = (io) => {
...
+        if(!userId){        
+          io.to(socket.id).emit("new-notification", {
+            error: true,
+            message: "User ID not provided",
+          });}
+        const toSocketId = await getUserSocketId(userId);
+        io.to(toSocketId.socketId).emit("new-notification", {
+          error: false
+        });
+        // console.log('sent notification');
```
Comment (jacquelineyin) [inline socket/socket.js:112]: delete

File: socket/socket.js [hunk 3]
```diff
@@ -94,6 +94,31 @@ const socketEvents = (io) => {
...
+      } catch (error) {
+        // console.log(error);
```
Comment (jacquelineyin) [inline socket/socket.js:114]: delete

**Review by jacquelineyin — APPROVED** [review]

File: routes/messages.js [hunk 4]
```diff
@@ -89,4 +89,61 @@ router.post("/", async (req, res) => {
...
+    await Message.deleteMany({});
+    res.status(204).send();
+  } catch (err) {
+    res.status(500).json({
+      message: "Error code 500: Failed to process request",
+    });
+  }
+});
+
+router.get("/unread/:userId", async (req, res) => {
```
Comment (Cainuse) [inline routes/messages.js:103]: Would prefer "/:userId/unread" to stay consistent with postings

File: routes/messages.js [hunk 5]
```diff
@@ -89,4 +89,61 @@ router.post("/", async (req, res) => {
   }
 });
 
+router.delete("/", async (req, res) => {
```
Comment (Cainuse) [inline routes/messages.js:92]: This endpoint is kind of dangerous, will we ever use this in production context?

**Comment by Cainuse** [comment]
Oh... it got merged while I was reviewing it lol 
I just left some nit-picky comments so no big deal

**Comment by abid-salahi** [comment]
Its ok we'll do code clean up once all the functionality code is in

**Pull request merged**